### PR TITLE
metastore dataproc service data source

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes https://github.com/hashicorp/terraform-provider-google/issues/10462

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
dataproc_metastore_service
```

### CODE REVIEW

**Comment by megan07** [comment]
Hi @DrFaust92 , I had a chance to review this and run the tests locally. I think the only things missing here would be to add this datasource to the provider: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/provider.go.erb#L259 and add some documentation. Thanks!

**Comment by DrFaust92** [comment]
Thanks @megan07, made some change and tested now.

--- PASS: TestAccDataprocMetastoreServiceDatasource_basic (861.13s)

File: mmv1/third_party/terraform/tests/data_source_dataproc_metastore_service_test.go.erb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataprocMetastoreServiceDatasource_basic(t *testing.T) {
+	t.Parallel()
+
+	name := "tf-metastore-" + randString(t, 10)
```
Comment (megan07) [inline mmv1/third_party/terraform/tests/data_source_dataproc_metastore_service_test.go.erb:15]: Sorry! I missed this yesterday. Would you mind changing the name here to begin with `tf-test` please? We have sweepers that will clean up old test resources that may linger that are prefixed with that. Thanks!

**Review by megan07 — APPROVED** [review]
Looks great and tests pass! If you wouldn't mind changing that test name, that'd be great! Thanks!

File: mmv1/third_party/terraform/tests/data_source_dataproc_metastore_service_test.go.erb (hunk 1 above)
Comment (DrFaust92) [reply mmv1/third_party/terraform/tests/data_source_dataproc_metastore_service_test.go.erb]: @megan07 done 👍

**Pull request merged**